controllers/challenge_controller_test.go

```diff
@@ -0,0 +1,223 @@
+				"description": "Description",
+				"points": 10
+			}`)
+
+			// Test the first challenge, and that the response is OK.
+			challengeController.Create(context)
+			Expect(context.Writer.Status()).To(Equal(http.StatusOK))
+
+			// Test that the challenge was written to mongo.
+			challenge := &models.Challenge{}
+			_ = mgm.Coll(challenge).First(bson.M{}, challenge)
+			Expect(challenge).NotTo(BeNil())
+
+			// Attempt creating a user with just the same location.
+			context2 := helpers.CreateTestContext("", `{
+				"latitude": 34.0701449,
+				"longitude": -118.4422936,
+				"address": "1234 Main Street",
+				"taskName": "Task 2",
+				"description": "Description",
+				"points": 20
+			}`)
+			challengeController.Create(context2)
+			Expect(context2.Writer.Status()).To(Equal(http.StatusOK))
+
+			// Test that the challenge was written to mongo.
+			_ = mgm.Coll(challenge).First(bson.M{}, challenge)
+			Expect(challenge).NotTo(BeNil())
+
+			// Attempt creating a user with just the same task name.
+			context3 := helpers.CreateTestContext("", `{
+				"latitude": 20.02,
+				"longitude": 20.02,
+				"address": "1234 Main Street",
+				"taskName": "Task 1",
+				"description": "Description",
+				"points": 20
+			}`)
+			challengeController.Create(context3)
+			Expect(context3.Writer.Status()).To(Equal(http.StatusOK))
+
+			// Test that the challenge was written to mongo.
+			_ = mgm.Coll(challenge).First(bson.M{}, challenge)
+			Expect(challenge).NotTo(BeNil())
+
+			// Attempt creating a challenge with the same location & taskName.
+			context4 := helpers.CreateTestContext("", `{
+				"latitude": 34.0701449,
+				"longitude": -118.4422936,
+				"address": "1234 Main Street",
+				"taskName": "Task 1",
+				"description": "Description",
+				"points": 20
+			}`)
+			challengeController.Create(context4)
+			Expect(context4.Writer.Status()).To(Equal(http.StatusBadRequest))
+
+			// Attempt creating challenge with no location/address info, but same task name
+			context5 := helpers.CreateTestContext("", `{
+				"taskName": "Task 1",
+				"description": "Description",
+				"points": 20
+			}`)
+			challengeController.Create(context5)
+			Expect(context5.Writer.Status()).To(Equal(http.StatusBadRequest))
+
+		}))
+
+		It("Test No Location Info", helpers.TestWithMongo("ChallengeController-Create-No-Location", func() {
+
+			// Attempt creating a challenge with only the minimum required fields.
+			context := helpers.CreateTestContext("", `{
+				"taskName": "Task 1",
+				"description": "Description",
+				"points": 20
+			}`)
+			challengeController.Create(context)
+			Expect(context.Writer.Status()).To(Equal(http.StatusOK))
+
+			// Test that the challenge was written to mongo.
+			challenge := &models.Challenge{}
+			_ = mgm.Coll(challenge).First(bson.M{}, challenge)
+			Expect(challenge).NotTo(BeNil())
+
+			// Attempt creating a challenge with the same taskName.
+			duplicateContext := helpers.CreateTestContext("", `{
```

Comment:
hmm maybe this isn't needed, since you already test for uniqueness calls above?

Reply:
removed extra test